StyleCop.Analyzers/StyleCop.Analyzers/Settings/SettingsHelper.cs

```diff
@@ -167,5 +228,39 @@ private static StyleCopSettings GetSettings(AnalyzerOptions options, SyntaxTree
 
             return new StyleCopSettings();
         }
+
+        private static SettingsFile GetSettingsFile(AnalyzerOptions options, Func<SourceText, Lazy<JsonValue>> getJsonValue, CancellationToken cancellationToken)
+        {
+            var additionalFiles = options != null ? options.AdditionalFiles : ImmutableArray.Create<AdditionalText>();
+            foreach (var additionalFile in additionalFiles)
+            {
+                if (IsStyleCopSettingsFile(additionalFile.Path))
+                {
+                    SourceText additionalTextContent = additionalFile.GetText(cancellationToken);
+                    var content = getJsonValue(additionalTextContent);
+                    return new SettingsFile(additionalFile.Path, content);
+                }
+            }
+
+            return null;
+        }
+
+        private static Lazy<JsonValue> ParseJson(SourceText text)
+        {
+            return new Lazy<JsonValue>(() => JsonReader.Parse(text.ToString()));
```

Comment:
💡 Would be easy to use [`SourceTextReader`](https://github.com/dotnet/roslyn-analyzers/commit/4369dd47748557d2e56161a783c749a020c7e83a) to avoid the call to `ToString()` here

Reply:
Handled in #3648